lib/components/date/DatePicker.tsx

```diff
@@ -0,0 +1,156 @@
+import { useState } from 'react'
+import { ArrowDown, ArrowUp, ChevronDown } from 'lucide-react'
+import type { Languages } from '../../hooks/useLanguage'
+import type { PropsWithLanguage } from '../../hooks/useTranslation'
+import { useTranslation } from '../../hooks/useTranslation'
+import { noop } from '../../util/noop'
+import { addDuration, monthsList, subtractDuration } from '../../util/date'
+import { tw, tx } from '../../twind'
+import type { YearMonthPickerProps } from './YearMonthPicker'
+import { YearMonthPicker } from './YearMonthPicker'
+import type { DayPickerProps } from './DayPicker'
+import { DayPicker } from './DayPicker'
+
+type DatePickerTranslation = {
+  year: string,
+  month: string,
+  day: string,
+  january: string,
+  february: string,
+  march: string,
+  april: string,
+  may: string,
+  june: string,
+  july: string,
+  august: string,
+  september: string,
+  october: string,
+  november: string,
+  december: string
+}
+
+const defaultDatePickerTranslation: Record<Languages, DatePickerTranslation> = {
+  en: {
+    year: 'Year',
+    month: 'Month',
+    day: 'Day',
+    january: 'January',
+    february: 'Febuary',
+    march: 'March',
+    april: 'April',
+    may: 'May',
+    june: 'June',
+    july: 'July',
+    august: 'August',
+    september: 'September',
+    october: 'October',
+    november: 'November',
+    december: 'December',
+  },
+  de: {
+    year: 'Jahr',
+    month: 'Monat',
+    day: 'Tag',
+    january: 'Januar',
+    february: 'Febuar',
+    march: 'März',
+    april: 'April',
+    may: 'Mai',
+    june: 'Juni',
+    july: 'Juli',
+    august: 'August',
+    september: 'September',
+    october: 'October',
+    november: 'November',
+    december: 'December',
+  }
+}
+
+type DisplayMode = 'yearMonth' | 'day'
+
+export type DatePickerProps = {
+  value?: Date,
+  start?: Date,
+  end?: Date,
+  initialDisplay?: DisplayMode,
+  onChange?: (date: Date) => void,
+  dayPickerProps?: Omit<DayPickerProps, 'value'|'onChange'|'selected'>,
+  yearMonthPickerProps?: Omit<YearMonthPickerProps, 'value'|'onChange'|'startYear'|'endYear'>,
+  className?: string
+}
+
+/**
+ * A Component for picking a date
+ */
+export const DatePicker = ({
```

Comment:
<img width="245" alt="image" src="https://github.com/helpwave/web/assets/16372321/cd217405-1076-4a26-819c-0bed50501c5d">

I'd make the `26` until `29` a bit lighter

Reply:
One more thing: I feel like the hover boxes should all have equal size (thus not depending on the width of the text)
It might also make sense to use a monospace font for the numbers, i cannot really tell right now if this is a good idea or not but worth a shot

Reply:
All boxes have equal size it just does not appear that way because of the font and number of digits, using a monospace font and padded 0s could be a solution in the future